src/GraphQL/Attributes/DirectiveAttribute.cs

```diff
@@ -0,0 +1,91 @@
+using GraphQL.Types;
+using GraphQL.Utilities;
+
+namespace GraphQL.Attributes;
+
+/// <summary>
+/// Applies a directive to part of a schema.
+/// </summary>
+[AttributeUsage(AttributeTargets.Enum | AttributeTargets.Class | AttributeTargets.Interface | AttributeTargets.Struct | AttributeTargets.Property | AttributeTargets.Method | AttributeTargets.Field | AttributeTargets.Parameter, AllowMultiple = true)]
+public class DirectiveAttribute : GraphQLAttribute
+{
+    /// <inheritdoc cref="DirectiveAttribute"/>
+    public DirectiveAttribute(string name)
+    {
+        Name = name;
+    }
+
+    /// <inheritdoc cref="DirectiveAttribute"/>
+    public DirectiveAttribute(string name, string argumentName, object argumentValue)
+    {
+        Name = name;
+        Arguments.Add(argumentName, argumentValue);
+    }
+
+    /// <inheritdoc cref="DirectiveAttribute"/>
+    public DirectiveAttribute(string name, string argumentName1, object argumentValue1, string argumentName2, object argumentValue2)
+    {
+        Name = name;
+        Arguments.Add(argumentName1, argumentValue1);
+        Arguments.Add(argumentName2, argumentValue2);
+    }
+
+    /// <inheritdoc cref="DirectiveAttribute"/>
+    public DirectiveAttribute(string name, string argumentName1, object argumentValue1, string argumentName2, object argumentValue2, string argumentName3, object argumentValue3)
+    {
+        Name = name;
+        Arguments.Add(argumentName1, argumentValue1);
+        Arguments.Add(argumentName2, argumentValue2);
+        Arguments.Add(argumentName3, argumentValue3);
+    }
+
+    /// <inheritdoc cref="DirectiveAttribute"/>
+    /// <remarks>
+    /// The <paramref name="argsAndValues"/> parameter must contain an even number of elements, where
+    /// the first element of each pair is the argument name and the second element is the argument value.
+    /// </remarks>
+    public DirectiveAttribute(string name, params object[] argsAndValues)
```

Comment:
Is it worth adding an analyzer to ensure an even number of elements and the argument names are strings?

Reply:
Probably pretty low on the priority list.  Sure that would be nice, and it could check that the first element of each pair was a string too.  Keep people from doing stuff like `[Directive("maxlength", 5)]` when they meant to do `[Directive("maxlength", "value", 5)]`